languages/csharp/exercises/concept/resource-lifetime/.docs/after.md

```diff
@@ -0,0 +1,67 @@
+We discussed in (TODO cross-ref-tba) how the `IDispoable` interface helps signal to callers of a class that there are resources or program state that need releasing or resetting in a timely fashion when the object in question is no longer required. In this exercise we have introduced some syntactic sugar, with the `using` keyword, that makes the code less verbose and less likely that significant calls will be omitted.
+
+`using` can be seen as replacing [`try/finally`][try-finally] for some use cases.
+
+```csharp
+File file = null;
+try
+{
+    file = new File("myStuff.txt");
+    file.Write("more stuff");
+}
+finally
+{
+    file.Dispose();
+}
+```
+
+This syntax is replaced with the more compact and foolproof:
+
+```csharp
+using (var file = new File("myStuff.txt")
+{
+    file.Write("more stuff");
+}
+```
```

Comment:
Do you think it would be useful to also note that you can have multiple `using` statements that have the same indentation if you have.

So instead of:

```csharp
using (var file = new File("mystuff.txt")
{
    using (var fileWriter= new StreamWriter(file)
    { 
         ...
     }
}
```

that one can do:

```csharp
using (var file = new File("mystuff.txt")
using (var fileWriter = new StreamWriter(file)
{ 
     ...
}
```

I've used this a lot when working with multiple dependent disposables.

Reply:
Done